### PR TITLE
first push of compition and result class

### PR DESCRIPTION
**Pull request by Moshizzl3**
What kind of change does this PR introduce?

- [ ] `bug-fix`
- [X] `new-feature`
- [ ] `enhancement`

### CODE REVIEW

File: src/main/java/com/app/controllers/CompetitionController.java [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+public class CompetitionController {
+
+  ArrayList<CompetitionModel> competitions;
+
+  public void createNewCompetition() {
+    // TODO we need to get some clarification regarding how we read input
+  }
+
+  /**
+   * a method that returns a competition based on the ID provided
```
Comment (MathiasReker) [inline src/main/java/com/app/controllers/CompetitionController.java]:
```suggestion
   * Returns a competition based on the provided ID.
```

File: src/main/java/com/app/controllers/CompetitionController.java [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    for (CompetitionModel competition : competitions) {
+      if (id.equals(competition.getId())) {
+        return competition;
+      }
+    }
+    return null;
+  }
+
+  /**
+   * A method that adds a result, to a given competition
```
Comment (MathiasReker) [inline src/main/java/com/app/controllers/CompetitionController.java]:
```suggestion
   * Add the result to a given competition.
```

File: src/main/java/com/app/models/ResultModel.java [hunk 3]
```diff
@@ -1,3 +1,44 @@
...
+
+  public LocalTime getResultTime() {
+    return resultTime;
+  }
+
+  public DisciplineModel getDiscipline() {
+    return discipline;
+  }
+
+  // setter
```
Comment (MathiasReker) [inline src/main/java/com/app/models/ResultModel.java]:
```suggestion
```

File: src/main/java/com/app/models/ResultModel.java [hunk 4]
```diff
@@ -1,3 +1,44 @@
...
+  private LocalTime resultTime;
+  private DisciplineModel discipline;
+
+  public ResultModel(MemberModel member, LocalTime resultTime, DisciplineModel discipline) {
+    this.member = member;
+    this.resultTime = resultTime;
+    this.discipline = discipline;
+  }
+
+  // getter
```
Comment (MathiasReker) [inline src/main/java/com/app/models/ResultModel.java]:
```suggestion
```

File: src/main/java/com/app/models/CompetitionModel.java [hunk 5]
```diff
@@ -1,3 +1,69 @@
...
+
+  public LocalTime getStartTime() {
+    return startTime;
+  }
+
+  public ArrayList<ResultModel> getResult() {
+    return result;
+  }
+
+  // setter
```
Comment (MathiasReker) [inline src/main/java/com/app/models/CompetitionModel.java]:
```suggestion
```

File: src/main/java/com/app/models/CompetitionModel.java [hunk 6]
```diff
@@ -1,3 +1,69 @@
...
+  private ArrayList<ResultModel> result = new ArrayList<>();
+
+  public CompetitionModel(LocalDate startDate, String name, LocalTime startTime) {
+    setId("test");
+    this.startDate = startDate;
+    this.name = name;
+    this.startTime = startTime;
+  }
+
+  // getter
```
Comment (MathiasReker) [inline src/main/java/com/app/models/CompetitionModel.java]:
```suggestion
```

File: src/main/java/com/app/controllers/CompetitionController.java [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+    DateTimeFormatter dateFormatter = DateTimeFormatter.ofPattern(timeFormat);
+
+    LocalTime time = null;
+
+    try {
+      time = LocalTime.parse(timeString, dateFormatter);
+    } catch (DateTimeParseException e) {
+      e.printStackTrace();
+    }
+    return time;
```
Comment (MathiasReker) [inline src/main/java/com/app/controllers/CompetitionController.java:85]: rename `time` to `result`

File: src/main/java/com/app/controllers/CompetitionController.java [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+    DateTimeFormatter dateFormatter = DateTimeFormatter.ofPattern(dateFormat);
+
+    LocalDate date = null;
+
+    try {
+      date = LocalDate.parse(dateString, dateFormatter);
+    } catch (DateTimeParseException e) {
+      e.printStackTrace();
+    }
+    return date;
```
Comment (MathiasReker) [inline src/main/java/com/app/controllers/CompetitionController.java:64]: rename `date` to `result`

**Review by Jarkyman — APPROVED** [review]

File: src/main/java/com/app/controllers/CompetitionController.java [hunk 9]
```diff
@@ -0,0 +1,87 @@
...
+
+  /**
+   * a method that returns a competition based on the ID provided
+   *
+   * @param id is the competition id for the competition you wish to return
+   * @return a competition based on the id that is provided
+   */
+  public CompetitionModel getCompetition(String id) {
+
+    for (CompetitionModel competition : competitions) {
```
Comment (MathiasReker) [inline src/main/java/com/app/controllers/CompetitionController.java:28]: rename `competition` to `c`

File: src/main/java/com/app/models/CompetitionModel.java [hunk 10]
```diff
@@ -1,3 +1,69 @@
...
+public class CompetitionModel {
+
+  private String id;
+  private LocalDate startDate;
+  private String name;
+  private LocalTime startTime;
+  private ArrayList<ResultModel> result = new ArrayList<>();
+
+  public CompetitionModel(LocalDate startDate, String name, LocalTime startTime) {
+    setId("test");
```
Comment (MathiasReker) [inline src/main/java/com/app/models/CompetitionModel.java]:
```suggestion
    setId("test"); // TODO
```

**Review by MathiasReker — CHANGES_REQUESTED** [review]

File: src/main/java/com/app/controllers/CompetitionController.java (hunk 9 above)
Comment (Andreassim) [reply src/main/java/com/app/controllers/CompetitionController.java:28]: I dont think its that important. Move it to a low priority issue.

File: src/main/java/com/app/controllers/CompetitionController.java (hunk 8 above)
Comment (Andreassim) [reply src/main/java/com/app/controllers/CompetitionController.java:64]: I dont think its that important. Move it to a low priority issue.

File: src/main/java/com/app/models/CompetitionModel.java (hunk 10 above)
Comment (Andreassim) [reply src/main/java/com/app/models/CompetitionModel.java]: Add this and we can merge

**Review by Andreassim — COMMENTED** [review]
There is only the todo in line 16 in CompetitonModel to add

**Review by Andreassim — APPROVED** [review]

**Pull request merged**